R/simple_ensemble.R

```diff
@@ -38,24 +38,53 @@
 #'   calculation issue that results in invalid distributions.
 #'
 #' @return a `model_out_tbl` object of ensemble predictions. Note that
-#'   any additional columns in the input `model_outputs` are dropped.
+#'   any additional columns in the input `model_out_tbl` are dropped.
 #'
 #' @export
-simple_ensemble <- function(model_outputs, weights = NULL,
+#'
+#' @examples
+#' # Calculate a weighted median in two ways
+#' model_outputs <- expand.grid(stringsAsFactors = FALSE,
+#'                              model_id = letters[1:4],
+#'                              location = c("222", "888"),
+#'                              horizon = 1, #week
+#'                              target = "inc death",
+#'                              target_date = as.Date("2021-12-25"),
+#'                              output_type = "quantile",
+#'                              output_type_id = c(.1, .5, .9),
+#'                              value = NA_real_)
+#'
+#' model_outputs$value <- c(10, 30, 15, 20, 100, 300, 400, 250, 40, 40, 45, 50,
+#'                          150, 325, 500, 300, 60, 70, 75, 80, 250, 350, 500, 350)
+#'
+#' fweights <- expand.grid(stringsAsFactors = FALSE,
+#'                         model_id = letters[1:4],
+#'                         location = c("222", "888"),
+#'                         weight = NA_real_)
+#' fweights$weight <- c(0.1 * (1:4), 0.1 * (4:1))
```

Comment:
Thank you for taking the time to make this example. One thing that strikes me is the fact that the setup takes three times the number of lines as the actual example---it's a common problem, so much so that it gets a specific mention in the R packages book:

>  If you find yourself needing to do a bunch of setup to create a dataset or object to use in the example, it may be a sign that you need to create a package dataset or even a helper function. See Chapter 7, Section 7.3.2, and Section 15.1.1 for ideas. Making it easy to write (and read) examples will greatly improve the quality of your documentation.
> --- [Wickham and Bryan, 2023](https://r-pkgs.org/man.html#contents)

I know this may be piling more work on to you, and I would be happy to take it on, but I see three options here:

1. leave it as is
   - pro: the example works! why change?
   - con: it can be distracting for both users and future developers (this includes future you to figure out what's happening in this example)
2. add comments to the generating code to clearly indicate that this is creating an example data set
3. move this code into [a script in a `data-raw/` folder](https://r-pkgs.org/data.html#sec-data-data-raw) to generate the data using `usethis::use_data_raw()`.
   - pro: the example now centers `simple_ensemble()`
   - con: up-front work to call `usethis::use_data_raw()`, copy over the script, run it, and then commit it. 

I would suggest to go with option 3 and clearly label these data sets as examples.

Reply:
NOTE: this suggestion also applies to the example in `linear_pool()`.